map.json

```diff
@@ -0,0 +1,117 @@
+[
+    {
+        "version": 2,
+        "path": "nvram3.bin",
+        "entries": [
+            {
+                "name": "_version",
+                "offset": 53,
+                "size": 1
+            },
+            {
+                "name": "save3_bool",
+                "offset": 54,
+                "size": 1,
+                "bit_offset": 0,
+                "is_bool": true
```

Comment:
I'd say always consider bits, but default them to `bit_offset=0` and `bit_size=8`. This way there is no need to check for `is_bool` and it will work in cases where one is storing packed data such as:
```c
struct my_struct {
    uint8_t a_bit:1;
    uint8_t b_value:7;
};
```

Then you specify `bit_offset` is an offset to the left of the starting byte offset and that `bit_size` must be smaller than `size * 8`. This way you can generically encode numbers in any amount of bits. Of course take care to unpack/pack those (as you did in IIO's code).